doc/index.md

```diff
@@ -0,0 +1,154 @@
+# Documentation for mlpack
+
+## A fast, flexible machine learning library
+
+mlpack is an intuitive, fast, and flexible header-only C++ machine learning
+library with bindings to other languages.  It aims to provide fast, lightweight
+implementations of both common and cutting-edge machine learning algorithms.
+
+mlpack's lightweight C++ implementation makes it ideal for deployment, and it
```

Comment:
```suggestion
mlpack's [lightweight C++ implementation](quickstart/cpp.md) makes it ideal for deployment, and it
```

Reply:
I did it in a slightly different place (closer to the binding quickstart links) in 680fee01be ---thanks for the suggestion. :+1: